crates/accelerate/src/linear_matrix.rs

```diff
@@ -0,0 +1,102 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use ndarray::{s, ArrayViewMut2, Axis};
+use numpy::PyReadwriteArray2;
+use pyo3::prelude::*;
+
+// Perform ROW operation on a matrix mat
+fn _row_op(mat: &mut ArrayViewMut2<bool>, ctrl: usize, trgt: usize) {
+    let row0 = mat.row(ctrl).to_owned();
+    let mut row1 = mat.row_mut(trgt);
+    row1.zip_mut_with(&row0, |x, &y| *x ^= y);
+}
+
+// Perform COL operation on a matrix mat
+fn _col_op(mat: &mut ArrayViewMut2<bool>, ctrl: usize, trgt: usize) {
+    let col0 = mat.column(ctrl).to_owned();
+    let mut col1 = mat.column_mut(trgt);
+    col1.zip_mut_with(&col0, |x, &y| *x ^= y);
+}
+
+// Gauss elimination of a matrix mat with m rows and n columns.
+// If full_elim = True, it allows full elimination of mat[:, 0 : ncols]
+// Returns the matrix mat.
+fn gauss_elimination(mut mat: ArrayViewMut2<bool>, ncols: Option<usize>, full_elim: Option<bool>) {
```

Comment:
No comment, just a question: why is the `mut` needed here? Don't we already have a mutable reference because of the `ArrayViewMut` type?

Reply:
otherwise I get compilation errors: " cannot borrow as mutable"